src/domains/citizen-auth-common/03_storage.tf

```
@@ -277,3 +277,25 @@ resource "azurerm_storage_queue" "profiles_to_sanitize" {
   name                 = "profiles-to-sanitize"
   storage_account_name = module.io_citizen_auth_storage.name
 }
+
+
+module "azure_storage_account" {
```

I would give the module a more talking name. Most importantly when other storage accounts are defined in the same terraform configuration.

```suggestion
module "io_citizen_auth_storage_itn" {
```